nomenclature/codelist.py

```diff
@@ -100,6 +100,25 @@ def check_variable_region_aggregation_args(cls, values):
                     )
         return values
 
+    @root_validator(pre=False, skip_on_failure=True)
+    def cast_variable_components_args(cls, values):
+        """Cast "components" list of dicts to a codelist"""
+        if values["name"] == "variable":
+            items = [
```

Comment:
Alternatively, you could iterate over `values["mapping"].items()` once, and check for each loop iteration if the attributes contain `"components"` instead of creating the items variable and iterating over it again.
Might be more easily understood this way though and the performance difference should be negligible.

Reply:
Good suggestion, implemented.

Reply:
I was worried that this would create a concurrent-edit-issue...